crowbar_framework/app/models/ipmi_service.rb

```diff
@@ -62,14 +63,16 @@ def transition(inst, name, state)
     #
     if state == "discovered"
       @logger.debug("IPMI transition: discovered state for #{name}")
-      db = Proposal.where(barclamp: "ipmi", name: inst).first
       role = RoleObject.find_role_by_name "ipmi-config-#{inst}"
-      result = add_role_to_instance_and_node("ipmi", inst, name, db, role, "ipmi-configure")
-
       node = NodeObject.find_node_by_name(name)
-      # Add the bmc routing roles as appropriate.
-      bmc_role = node.admin? ? "bmc-nat-router" : "bmc-nat-client"
-      result = add_role_to_instance_and_node("ipmi", inst, name, db, role, bmc_role)
+
+      Proposal.where(barclamp: "ipmi", name: inst).first do |db|
+        result = add_role_to_instance_and_node("ipmi", inst, name, db, role, "ipmi-configure")
+
+        # Add the bmc routing roles as appropriate.
+        bmc_role = node.admin? ? "bmc-nat-router" : "bmc-nat-client"
+        result = add_role_to_instance_and_node("ipmi", inst, name, db, role, bmc_role)
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `result`. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)